use the `run` command of `conda` / `mamba` / `micromamba` and support `pixi`

keewis opened the pull request:
## Change Summary

This adds the `--environment-manager` and `--environment-manager-path` cli options, as well as replacing the `activate` step with the `run` subcommand.

No tests, yet.

## Related issue number

- [x] closes #257, closes #248

## Checklist

- [ ] Unit tests for the changes exist
- [x] Tests pass on CI
- [ ] Documentation reflects the changes where applicable

keewis committed:
add an option to configure the env manager

keewis committed:
add a helper for detecting paths

keewis committed:
allow customizing the path to the env manager

keewis committed:
use `run` instead of activating the environment

keewis committed:
remove the activation logic

keewis committed:
ensure pixi works properly

keewis committed:
remove the debug rule preventing the actual execution

keewis committed:
Merge branch 'main' into run

keewis committed:
factor out the creation of templates

keewis committed:
infer the environment manager

keewis committed:
Merge branch 'main' into run

keewis committed:
remove the `conda activate` tests

keewis committed:
pass a env manager to `_prepare_batch_job_script`

keewis committed:
use a fixture instead

keewis committed:
instantiate

keewis committed:
properly construct the environment manager

keewis committed:
set the conda env

keewis committed:
actually check the script file

keewis committed:
also check that the environment is being used

keewis committed:
better documentation for the new options

andersy005 reviewed (approved):
thank you so much for putting this together, @keewis! It looks solid

keewis committed:
add `--frozen` to the script template of `pixi`

lots of compute nodes don't have internet access, so this can improve
the startup times a bit

keewis committed:
simple implementation for the uv environment manager

keewis committed:
reuse the actual manager names

keewis commented:
@andersy005, any chance you could have another look? (No pressure if you've been away / busy, though, I just want to make sure this doesn't get forgotten)

I've addressed the `uv` part (partially, at least) and have been using the pixi integration on my local hpc quite a bit, so I know this should work for most use-cases

keewis commented:
... and immediately after the ping a colleague of mine noticed issues with the way `which jupyter` is called. I'll retract my request for a review until I've found the time to verify why this fails
